src/emit/mod.rs

```diff
@@ -3247,6 +3251,68 @@ pub trait TargetRuntime<'a> {
         function: FunctionValue<'a>,
         ns: &Namespace,
     ) {
+        let dibuilder = &bin.dibuilder;
+        let compile_unit = &bin.compile_unit;
+        let file = compile_unit.get_file();
+        let mut di_func_scope: Option<DISubprogram<'_>> = None;
+
+        let return_type = function.get_type().get_return_type();
+        match return_type {
+            None => {}
+            Some(return_type) => {
+                let return_type_size = return_type.size_of().unwrap();
+                let size = return_type_size.get_type().get_bit_width();
+                let mut type_name = "size_".to_owned();
+                type_name.push_str(&size.to_string());
+                let di_return_type = dibuilder
+                    .create_basic_type(
+                        &type_name,
+                        size as u64,
+                        0x00,
+                        inkwell::debug_info::DIFlagsConstants::PUBLIC,
+                    )
+                    .unwrap();
+                let param_types = function.get_type().get_param_types();
+                let di_param_types: Vec<DIType<'_>> = param_types
+                    .iter()
+                    .map(|typ| {
+                        let mut param_tname = "size_".to_owned();
+                        let param_size = typ.size_of().unwrap().get_type().get_bit_width();
+                        param_tname.push_str(&size.to_string());
+                        dibuilder
+                            .create_basic_type(
+                                &param_tname,
+                                param_size as u64,
+                                0x00,
+                                inkwell::debug_info::DIFlagsConstants::PUBLIC,
+                            )
+                            .unwrap()
+                            .as_type()
+                    })
+                    .collect();
+                let di_func_type = dibuilder.create_subroutine_type(
+                    file,
+                    Some(di_return_type.as_type()),
+                    di_param_types.as_slice(),
+                    inkwell::debug_info::DIFlagsConstants::PUBLIC,
+                );
+                di_func_scope = Some(dibuilder.create_function(
+                    compile_unit.as_debug_info_scope(),
+                    function.get_name().to_str().unwrap(),
+                    None,
+                    file,
+                    /* line no*/ 0, // TODO: rectify line name
+                    di_func_type,
+                    true,
+                    true,
+                    /* scope line */ 0, // TODO: rectify scope line
+                    inkwell::debug_info::DIFlagsConstants::PUBLIC,
```

Comment:
Not all functions are public, also TODO could be implemented

Reply:
@seanyoung Do you have any suggestions to define the `flags` field, i.e., it is `PUBLIC` or `PRIVATE`?

Reply:
if `cfg.public` is true, ` inkwell::debug_info::DIFlagsConstants::PUBLIC`, else private.

Reply:
Thanks. How about the `scope_line` and `line_no`? I try to use `bin.builder.get_current_debug_location().unwrap();` but I always get the line number of `3267742850`.

Reply:
Currently, I use the location of the first instruction: `cfg.blocks[0].instr.first().unwrap().loc()`. Free free to request changes if you feel it is not a suitable way.